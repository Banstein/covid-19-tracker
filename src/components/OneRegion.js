import React from 'react';
import PropTypes from 'prop-types';

function OneRegion({
  name, todayConfirmed, todayDeaths, todayOpenCases, todayRecovered,
}) {
  return (
    <li className="flex justify-between striped">
      <div className="region-name">
        <h5>{name}</h5>
      </div>
      <div className="country-stats">
        <h6>
          <span>Confirmed: </span>
          {todayConfirmed}
        </h6>
        <h6>
          <span>Deaths: </span>
          {todayDeaths}
        </h6>
        <h6>
          <span>Open Cases: </span>
          {todayOpenCases}
        </h6>
        <h6>
          <span>Recovered: </span>
          {todayRecovered}
        </h6>
      </div>
    </li>
  );
}

OneRegion.propTypes = {
  name: PropTypes.string.isRequired,
  todayConfirmed: PropTypes.number.isRequired,
  todayDeaths: PropTypes.number.isRequired,
  todayOpenCases: PropTypes.number.isRequired,
  todayRecovered: PropTypes.number.isRequired,
};

export default OneRegion;
