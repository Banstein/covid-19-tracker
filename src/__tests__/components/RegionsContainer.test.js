/* eslint-disable camelcase */
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import store from '../../__mocks__/configureStore';
import OneRegion from '../../components/OneRegion';

describe('Countries snapshot', () => {
  it('renders correctly', () => {
    const id = 'germany';
    const name = 'Germany';
    const today_confirmed = 56;
    const today_deaths = 55;
    const today_open_cases = 54;
    const today_recovered = 53;
    render(
      <Provider store={store}>
        <BrowserRouter>
          <OneRegion
            id={id}
            name={name}
            todayConfirmed={today_confirmed}
            todayDeaths={today_deaths}
            todayOpenCases={today_open_cases}
            todayRecovered={today_recovered}
          />
        </BrowserRouter>
      </Provider>,
    );
  });
});
