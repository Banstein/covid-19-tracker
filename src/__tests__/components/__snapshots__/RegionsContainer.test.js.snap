// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Countries snapshot renders correctly 1`] = `
<li
  className="flex justify-between striped"
>
  <div
    className="region-name"
  >
    <h5>
      Germany
    </h5>
  </div>
  <div
    className="country-stats"
  >
    <h6>
      <span>
        Confirmed: 
      </span>
      56
    </h6>
    <h6>
      <span>
        Deaths: 
      </span>
      55
    </h6>
    <h6>
      <span>
        Open Cases: 
      </span>
      54
    </h6>
    <h6>
      <span>
        Recovered: 
      </span>
      53
    </h6>
  </div>
</li>
`;
